### ISSUE TITLE
Support costum xml tags (Origin: bugzilla #678473)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#4765]
status NEW severity *enhancement* in component *documentation* for *---*
Reported in version *1.8.2-SVN* on platform *Other*
Assigned to: Dimitri van Heesch

On 2012-06-20 12:01:19 +0000, Marcel Hellwig wrote:

> It is already possible to add Aliases - thats great. But in the last time I develop some applications with C# and they use the xml documentation format. Unfortunately you can't declare costum xml tags.
> 
> Let's see, what you can do.
> 
> Regards

On 2012-11-18 11:07:11 +0000, Dimitri van Heesch wrote:

> Changed version 'latest' to '1.8.2-SVN' so I can remove 'latest' as an option as it is a moving target.

**Comment by Arcturuss** [comment]
+1 for this request.

Some tools like Swagger rely on custom XML tags in comments, so it would be nice to have Doxygen understand them too.

Example here: https://stackoverflow.com/questions/54469140/styling-encoded-text-in-doxygen-generated-html

**Comment by mtrevin93** [comment]
Also interested in this feature

**Comment by albert-github** [comment]
To be able to make an implementation of this enhancement it is good to have a specification.

An attempt:

How does a tag look like?
```
<tag [attribute="value"]*>[text]?</tag>
```
so we would need a setting like:
```
ALIASES_XML = tag=...
```

How could this look like:

- A small example:
  ```
  <code>
  the source code line 1
  ...
  the source code line last
  </code>
  ```
  with the 
  ```
  ALIASES = code="\code^^$text^^@endcode"
  ```
  we would get:
  ```
  @code
  the source code line 1
  ...
  the source code line last
  @endcode
  ```

- Another small example:
  ```
  <img src="application.jpg" width="10cm">My application"</img>
  ```
  with the
  ```
  ALIASES = img="@image html $src \"$text\" width=$width"
  ```
  we would get:
  ```
  @image html application.jpg "My application" width=10cm
  ```

A bit more formal:
```
<tag [$attributename="$attributevalue"]*>[$text]?</tag>
```
in the `ALIASES_XML` the `$text` will be replaced with the actual text and the specified `$attributename` will be
replaced by the specified `$attributevalue`.
Note that this value will be without the surrounding double or single quotes.

**Comment by mtrevin93** [comment]
Hello, thank you for responding!

Yes, ideally, I would like to be able to parse a format like this:

`
<tag [attribute=value]>[text]</tag>
`

Is there any way to capture the attribute/value key value pair and pass it as an argument to the alias, or something similar? For a usage like:

`
ALIASES += "tag{1}=\xrefitem tag[attribute1=value1]} "Tag" "Tags" "[text]"
`

or:

`
ALIASES += "tag{2}=\xrefitem tag[1=2]} "Tag" "Tags" "[text]"
`

It's not perfect since it could be tricky if I want multiple `[attribute=value]`. But maybe I could decide what `[attribute=value]` to use for each tag, and what order I must list them in, or only support one pair per tag.

To summarize, my goal would be to be able to create a custom XML tag, then perform Doxygen special commands based on the tag I chose. Ideally, I am also able to incorporate attributes & values into these tags, so that I could create a grouping of tags based on attribute. This allows me to use all of Doxygen's special commands, but still keep XML style formatting instead of having to split between XML and Doxygen's special syntax.

Sorry if my response is not perfect, I have only been using Doxygen a couple of days. Thank you again for your help!

**Comment by albert-github** [comment]
@mtrevin93 
I'm not getting what you want with your "tag" example, especially the part `[attribute1=value1]` / `[1=2]` in the `ALIASES` (due to the `[` / `]`  and also here is also just a `}` without a `{`.
`Why would you transform a `ALIASES_XML` setting to an `ALIASES` setting, why not directly code this from the `ALIASES_XML`.

**Comment by mtrevin93** [comment]
Apologies. I thought you were proposing transforming ALIASES_XML to ALIASES as what would have to be done to make this feature possible. Kindly disregard.

As long as it is possible to type something like this `<tag [attribute="value"]*>[text]?</tag>` and be able to perform Doxgen special  commands that reference the attribute/value pairs, that would be ideal, for me.
